mm2src/lp_swap/taker_swap.rs

```diff
@@ -1233,13 +1254,22 @@ impl TakerSwap {
         );
         let tx = match f.compat().await {
             Ok(t) => t,
-            Err(e) => {
+            Err(err) => {
+                if let Some(tx) = err.get_tx() {
+                    broadcast_p2p_tx_msg(
```

Comment:
`wait_for_tx_spend` doesn't broadcast transaction. It detects that specific output/transaction is spent by other party and returns the spending transaction. It already exists in the blockchain at this point so `broadcast_p2p_tx_msg` is not required here.